### PR TITLE
Remove firebase_project_location

### PR DESCRIPTION
**Pull request by rainshen49**
Throw an error when users try to use the `google_firebase_project_location` resource in the 5.0.0 version.

closes https://github.com/hashicorp/terraform-provider-google/issues/15763

If this PR is for Terraform, I acknowledge that I have:

- [x] Read [Write release notes](https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:breaking-change
firebase: removed `google_firebase_project_location`
```

### CODE REVIEW

File: mmv1/products/firebase/ProjectLocation.yaml [hunk 1]
```diff
@@ -39,16 +39,10 @@ skip_delete: true
 skip_sweeper: true
 deprecation_message: >-
   `google_firebase_project_location` is deprecated in favor of explicitly configuring `google_app_engine_application`
-  and `google_firestore_database`. This resource will be removed in the next major release of the provider.
-examples:
-  - !ruby/object:Provider::Terraform::Examples
-    name: 'firebase_project_location_basic'
```
Comment (roaks3) [inline mmv1/products/firebase/ProjectLocation.yaml:45]: Should we also delete `mmv1/templates/terraform/examples/firebase_project_location_basic.tf.erb`?

**Review by roaks3 — COMMENTED** [review]
Looks like we're seeing this error in the build:
```
google-beta/services/firebase/resource_firebase_project_location.go:68:2: config declared but not used
```

I think it is because the `config` var used by the create function normally is not being used anymore. Do you know if there is precedence for the pattern you're using here?

File: mmv1/products/firebase/ProjectLocation.yaml (hunk 1 above)
Comment (rainshen49) [reply mmv1/products/firebase/ProjectLocation.yaml:45]: Oops, yes thanks!

**Comment by rainshen49** [comment]
I don't find a precedent. I'm trying to follow https://developer.hashicorp.com/terraform/plugin/sdkv2/best-practices/deprecations#provider-data-source-or-resource-removal, which says to 

> remove all code associated with the deprecated data source or resource except for the schema and replace the Create and Read functions to always return an error

Since this resource doesn't have any attributes, I don't think anyone would be reading them, but the Create function should be blocked.

Is there a better way to do that than custom_create?

**Review by roaks3 — APPROVED** [review]
@c2thorn would you mind weighing in on this? I wasn't able to find any precedent for deprecating resources, but this approach makes sense to me and seems to match the guidance from HashiCorp.

**Comment by rainshen49** [comment]
Friendly ping @c2thorn

**Review by c2thorn — CHANGES_REQUESTED** [review]
For other removed resources (like `google_project_services` and [game services](https://github.com/GoogleCloudPlatform/magic-modules/pull/8669)) we just completely delete the related code and do not leave stubbed resources. It would be nice for users to get a specific message stating that the resource is no longer available, but as of now we prefer to just completely remove.

**Comment by c2thorn** [comment]
We should add the actual deprecation message and upgrade guide note in the `main` branch. Does a PR for that exist already? @rainshen49 
Also, mind updating the PR/changelog note to be specific about removing?

**Comment by rainshen49** [comment]
Updated PR description.
Yes, the deprecation message and migration guide PR has been merged https://github.com/GoogleCloudPlatform/magic-modules/pull/8301

**Comment by c2thorn** [comment]
I've added https://github.com/hashicorp/terraform-provider-google/issues/15763 for our tracking efforts. If there is a buganizer number, do you mind adding the b/##### to the issue? @rainshen49

**Review by c2thorn — APPROVED** [review]

**Pull request merged**